### DISCUSSION TITLE
plugin installation: out of memory

### DISCUSSION THREAD

**Discussion by stenootje** [combostrap/combo#73] [Q&A]
Hi,
context: new dokuwiki install, combo plugin active, php.ini: memory_limit = 1280M
action: installing a plugin
result: Fatal error: Out of memory (allocated 1054867456) (tried to allocate 4096 bytes) in <path to dokuwiki>\lib\plugins\combo\ComboStrap\ExecutionContext.php on line 223

context: new dokuwiki install, combo plugin inactive, php.ini: memory_limit = 128M
action: installing Plugin bootswrapper installed successfully
result: Plugin bootswrapper installed successfully

Who has experience similar problems? Even got situation where enabling & disabling plugins failed

Thx

**Comment by gerardnico** [comment]
Yeah, that's too bad.

Follow this article for the solution.
https://combostrap.com/support/how-to-resolve-the-memory-error-when-updating-q5ytih3s

Unfortunately, as you will read, I can't change anything.

**Reply by stenootje** [reply]
Thanks for this info which brings the problem beyond the scope of a plugin.

Dokuwiki has an automatic and manual install. But a manual uninstall of a plugin or template is not provided. Maybe a plugin should be able to indicate whether it is part of automatic install process. If not, its directories should be excluded from the scan of the update script of DokuWiki. I hope you reported this to Dokuwiki folks. It is in any way a blocker for future scalability of dokuwiki.

With your information in mind, I worked out another solution for me: 
1) I placed the combo-plugin outside the dokuwiki directory
2) I made a script link.bat which creates a symbolic link (windows) to the plugin dir of dokuwiki
3) I made a script unlink.bat which deletes this symbolic link (rmdir)
Now I can use dokuwiki without combo (in case I get errors when combo is available) so I can install/update/enable/delete plugins from the admin pages.

**Reply by gerardnico** [reply]
Ouah cool. 
I wonder if you would get the same effect by just disabling the `combo` plugin on the admin page before updating other plugins.

**Comment by stenootje** [comment]
Sometimes it is possible, but I have got situations where neither install/uninstall, enble/disable is possible.
I can extend this to other plugins in case of conflicts or errors.... so for me a strait forward solution
